internal/integration/unified/client_operation_execution.go

```diff
@@ -163,3 +166,329 @@ func executeListDatabases(ctx context.Context, operation *operation, nameOnly bo
+		t, d, err := bson.MarshalValue(v.InsertedID)
+		if err != nil {
+			return nil, err
+		}
+		resBuilder.AppendDocument(strconv.Itoa(k),
+			bsoncore.NewDocumentBuilder().
+				AppendValue("insertedId", bsoncore.Value{Type: bsoncore.Type(t), Data: d}).
+				Build(),
+		)
+	}
+	rawBuilder.AppendDocument("insertResults", resBuilder.Build())
+
+	resBuilder = bsoncore.NewDocumentBuilder()
+	for k, v := range res.UpdateResults {
+		b := bsoncore.NewDocumentBuilder().
+			AppendInt64("matchedCount", v.MatchedCount).
+			AppendInt64("modifiedCount", v.ModifiedCount)
+		if v.UpsertedID != nil {
+			t, d, err := bson.MarshalValue(v.UpsertedID)
+			if err != nil {
+				return nil, err
+			}
+			b.AppendValue("upsertedId", bsoncore.Value{Type: bsoncore.Type(t), Data: d})
+		}
+		resBuilder.AppendDocument(strconv.Itoa(k), b.Build())
+	}
+	rawBuilder.AppendDocument("updateResults", resBuilder.Build())
+
+	return newDocumentResult(rawBuilder.Build(), err), nil
+}
+
+func appendClientBulkWriteModel(key string, value bson.Raw, model *mongo.ClientWriteModels) error {
+	switch key {
+	case "insertOne":
+		namespace, m, err := createClientInsertOneModel(value)
+		if err != nil {
+			return err
+		}
+		ns := strings.SplitN(namespace, ".", 2)
+		model.AppendInsertOne(ns[0], ns[1], m)
+	case "updateOne":
+		namespace, m, err := createClientUpdateOneModel(value)
+		if err != nil {
+			return err
+		}
+		ns := strings.SplitN(namespace, ".", 2)
+		model.AppendUpdateOne(ns[0], ns[1], m)
+	case "updateMany":
+		namespace, m, err := createClientUpdateManyModel(value)
+		if err != nil {
+			return err
+		}
+		ns := strings.SplitN(namespace, ".", 2)
+		model.AppendUpdateMany(ns[0], ns[1], m)
+	case "replaceOne":
+		namespace, m, err := createClientReplaceOneModel(value)
+		if err != nil {
+			return err
+		}
+		ns := strings.SplitN(namespace, ".", 2)
+		model.AppendReplaceOne(ns[0], ns[1], m)
+	case "deleteOne":
+		namespace, m, err := createClientDeleteOneModel(value)
+		if err != nil {
+			return err
+		}
+		ns := strings.SplitN(namespace, ".", 2)
+		model.AppendDeleteOne(ns[0], ns[1], m)
+	case "deleteMany":
+		namespace, m, err := createClientDeleteManyModel(value)
+		if err != nil {
+			return err
+		}
+		ns := strings.SplitN(namespace, ".", 2)
+		model.AppendDeleteMany(ns[0], ns[1], m)
+	}
+	return nil
+}
+
+func createClientInsertOneModel(value bson.Raw) (string, *mongo.ClientInsertOneModel, error) {
+	var v struct {
+		Namespace string
+		Document  bson.Raw
+	}
+	err := bson.Unmarshal(value, &v)
+	if err != nil {
+		return "", nil, err
+	}
+	return v.Namespace, &mongo.ClientInsertOneModel{
+		Document: v.Document,
+	}, nil
+}
+
+func createClientUpdateOneModel(value bson.Raw) (string, *mongo.ClientUpdateOneModel, error) {
+	var v struct {
+		Namespace    string
+		Filter       bson.Raw
+		Update       interface{}
+		ArrayFilters []interface{}
+		Collation    *options.Collation
+		Hint         *bson.RawValue
+		Upsert       *bool
+	}
+	err := bson.Unmarshal(value, &v)
```

Comment:
Why do we need to define the extra fields? Can we unmarshal into an embedded `mongo.ClientUpdateOneModel`?

```go
var v struct {
	Namespace string
	mongo.ClientUpdateOneModel
}
```

Reply:
The spec tests put bulk write fields in parallel with the namespace. We cannot embed `mongo.ClientUpdateOneModel` without a nested level.